e2e/e2eutil/utils.go

```diff
@@ -111,6 +111,22 @@ func WaitForAllocRunning(t *testing.T, nomadClient *api.Client, allocID string)
 	})
 }
 
+func DeploymentsForJob(nomadClient *api.Client, jobID string) []*api.Deployment {
+	ds, _, err := nomadClient.Deployments().List(nil)
+	if err != nil {
+		return nil
```

Comment:
I would consider taking `t *testing.T` as an argument and then failing when `List` call fails.